src/toil/common.py

```diff
@@ -1014,26 +1017,55 @@ def _setupAutoDeployment(self, userScript=None):
             logger.debug('Injecting user script %s into batch system.', userScriptResource)
             self._batchSystem.setUserScript(userScriptResource)
 
-    def importFile(self, srcUrl, sharedFileName=None, symlink=False):
+    def importFile(self,
+                   srcUrl: str,
+                   sharedFileName: Optional[str] = None,
+                   symlink: bool = False) -> Optional[Union[FileID, str]]:
         """
         Imports the file at the given URL into job store.
 
         See :func:`toil.jobStores.abstractJobStore.AbstractJobStore.importFile` for a
         full description
         """
         self._assertContextManagerUsed()
+        srcUrl = self.sanity_check_uri_and_format(srcUrl, check_existence=True)
         return self._jobStore.importFile(srcUrl, sharedFileName=sharedFileName, symlink=symlink)
 
-    def exportFile(self, jobStoreFileID, dstUrl):
+    def exportFile(self, jobStoreFileID: Union[FileID, str], dstUrl: str) -> None:
         """
         Exports file to destination pointed at by the destination URL.
 
         See :func:`toil.jobStores.abstractJobStore.AbstractJobStore.exportFile` for a
         full description
         """
         self._assertContextManagerUsed()
+        dstUrl = self.sanity_check_uri_and_format(dstUrl)
         self._jobStore.exportFile(jobStoreFileID, dstUrl)
 
+    @staticmethod
+    def sanity_check_uri_and_format(uri: str, check_existence: bool = False) -> str:
```

Comment:
I'm going to come out against "sanity" checks; I don't think the term is appropriate and we don't want to scare people who aren't, strictly speaking, sane away from our project.

That's also not what this function does anyway; it returns a string. maybe it should be called `normalize_url` or `ensure_url` or `urlify_paths` or something.

Reply:
Well, I suppose `normalize_url` has a certain ring to it, even though "normalization" has always sounded like what would immediately follow a oppressively political fellow aiming a very powerful laser at someone they disagree with.